src/main/java/org/candlepin/subscriptions/marketplace/MarketplaceSubscriptionCollector.java

```diff
@@ -56,9 +59,19 @@ public MarketplaceSubscriptionCollector(SubscriptionService subscriptionService)
     }
 
     protected List<Subscription> filterNonApplicableSubscriptions(List<Subscription> subscriptions) {
+
+        Comparator<Subscription> byEffectiveStartDate =
+            Comparator.comparing(Subscription::getEffectiveStartDate);
+
         return subscriptions.stream()
             .filter(sub -> !Objects.isNull(sub.getExternalReferences()))
             .filter(sub -> !Objects.isNull(sub.getExternalReferences().get(IBMMARKETPLACE)))
+            .collect(Collectors.groupingBy(sub -> sub.getSubscriptionNumber() == null ?
```

Comment:
This uses Subscription.subscriptionNumber to group subscriptions and determine the ones with the latest date by subscription number.  Please let me know if subscriptionNumber is the right field.  If not, I'll change this.